Classes/Domain/Validator/LanguageSkillValidator.php

```diff
@@ -0,0 +1,38 @@
+<?php
+namespace PAGEmachine\Ats\Domain\Validator;
+
+use TYPO3\CMS\Extbase\Validation\Validator\AbstractValidator;
+
+/*
+ * This file is part of the PAGEmachine ATS project.
+ */
+
+/**
+ * LanguageSkillValidator
+ */
+class LanguageSkillValidator extends AbstractValidator
+{
+    public function isValid($languageSkill)
+    {
+
+        $error = false;
+
+        //Either dropdown language or text language must be set
+        if ($languageSkill->getLanguage() === null && empty($languageSkill->getTextLanguage())) {
+            $this->result->forProperty('language')->addError(new \TYPO3\CMS\Extbase\Validation\Error('You must enter a language, either via dropdown oder text field.', 1522232293));
```

Comment:
You can `use` the `Error` to shorten this a bit.

Reply:
I'd also change the message to _... either via selection or input._ (BTW: _oder_ ;-) )